File: exercises/df1c4e80-ef17-4045-b970-46a15c710a09/static-correctors/4b5edce8-5ba2-46b3-ac43-9389d563cc96/beansGame.pas
(**
 * @author Margarida Mamede
 *)


program beansGame;

var
   seq             : array of integer;
   seqLen, elem, i : integer;
   sum, scA, scB   : integer;


function scoreOfAlex( ) : integer;

var
   steps, left, right, i : integer;
   score                 : integer;
   minStep               : boolean;

begin
   steps := seqLen div 2;
   left := 0;
   right := seqLen - 1;
   score := 0;
   minStep := true;

   for i := 1 to steps do
      begin
         (* Alex selection *)
         if seq[left] > seq[right] then
            begin
               score := score + seq[left];
               left := left + 1
            end
         else 
            begin
               score := score + seq[right];
               right := right - 1
            end;

         (* Bela selection *)
         if minStep then
            begin
               if seq[left] < seq[right] then
                  left := left + 1
               else
                  right := right - 1
            end
         else 
            begin
               if seq[left] > seq[right] then
                  left := left + 1
               else
                  right := right - 1
            end;
         minStep := not minStep
      end;

   scoreOfAlex := score
end;


begin
   readln(seqLen);
   setlength(seq, seqLen);

   sum := 0;
   for i := 0 to seqLen - 1 do
      begin
         read(elem);
         seq[i] := elem;
         sum := sum + elem
      end;

   scA := scoreOfAlex();
   scB := sum - scA;

   if scA > scB then
      writeln('Alex ganha com ', scA, ' contra ', scB)
   else if scA < scB then
      writeln('Bela ganha com ', scB, ' contra ', scA)
   else
      writeln('Alex e Bela empatam a ', scA)
end.

File: exercises/df1c4e80-ef17-4045-b970-46a15c710a09/skeletons/18615fe9-6438-4bb6-8aae-d5e09902f3b4/beansGame.c
#include<stdio.h>

/**
 * @author Margarida Mamede
 */


int scoreOfAlex( int seq[],  int seqLen )
{
    int steps = seqLen / 2;
    int left = 0; 
    int right = seqLen - 1; 
    int score = 0;
    int minStep = 1;

    for ( int i = 0; i < steps; i++ ) 
    {
        // Alex selection
        if ( seq[left] > seq[right] ) 
        {
            score += seq[left];
            left++;
        }
        else 
        {
            score += seq[right];
            right--;
        }

        // Bela selection
        if ( minStep ) 
        {
            if ( seq[left] < seq[right] )
                left++;
            else
                right--;
        }
        else 
        {
            if ( seq[left] > seq[right] )
                left++;
            else
                right--;
        }
        minStep = !minStep;
    }
    return score;
}


int main( ) 
{
    int seqLen, elem, sum, scA, scB;

    scanf("%d", &seqLen);
    int seq[seqLen];

    sum = 0;
    for ( int i = 0; i < seqLen; i++ )
    {
        scanf("%d", &elem);
        seq[i] = elem;
        sum += elem;
    }

    scA = scoreOfAlex(seq, seqLen);
    scB = sum - scA;

    if ( scA > scB )
        printf("Alex ganha com %d contra %d\n", scA, scB);
    else if ( scA < scB )
        printf("Bela ganha com %d contra %d\n", scB, scA);
    else
        printf("Alex e Bela empatam a %d\n", scA);
    return 0;
}

